central/reprocessor/reprocessor.go

```diff
@@ -544,13 +544,13 @@ func (l *loopImpl) riskLoop() {
 		case <-l.stopSig.Done():
 			return
 		case <-l.deploymentRiskTicker.C:
-			l.deploymentRiskLock.Lock()
-			if l.deploymentRiskSet.Cardinality() > 0 {
-				// goroutine to ensure this is non-blocking.
-				go l.sendDeployments(l.deploymentRiskSet.AsSlice())
-				l.deploymentRiskSet.Clear()
-			}
-			l.deploymentRiskLock.Unlock()
+			concurrency.WithLock(&l.deploymentRiskLock, func() {
+				if l.deploymentRiskSet.Cardinality() > 0 {
+					// goroutine to ensure this is non-blocking.
+					go l.sendDeployments(l.deploymentRiskSet.AsSlice())
+					l.deploymentRiskSet.Clear()
+				}
+			})
```

Comment:
I think, that in cases like this, where the critical section ends at the end of the function, it is much cleaner to use
```go
l.deploymentRiskLock.Lock()
defer l.deploymentRiskLock.Unlock()
```

Reply:
Not sure: the critical section is buried in a few level of indentation and is in a `case` statement, so the fact that the critical section ends at the end of the function is incidental.

Reply:
I don't think the critical section ends at the end of the function and isn't ends at the end of the case. I prefer the clarity that this change provides